src/main/kotlin/org/sol4k/Transaction.kt

```diff
@@ -90,5 +97,94 @@ class Transaction(
         private const val RECENT_BLOCK_HASH_LENGTH = 32
         private const val PUBLIC_KEY_LENGTH = 32
         private const val SIGNATURE_LENGTH = 64
+
+        fun from(encodedTransaction: String): Transaction? {
+            return try {
+                var byteArray = Base64.getDecoder().decode(encodedTransaction)
+
+                // 1. remove signatures
+                val signaturesCount = Binary.decodeLength(byteArray)
```

Comment:
You can deconstruct it straight away here. EG:
```
val (len, byteArray) = Binary.decodeLength(byteArray)
```
There are a couple of places where you can use it.